src/rules/fileHeaderRule.ts

```diff
@@ -63,7 +71,9 @@ export class Rule extends Lint.Rules.AbstractRule {
             offset,
             (pos, end, kind) => text.substring(pos + 2, kind === ts.SyntaxKind.SingleLineCommentTrivia ? end : end - 2));
 
-        if (commentText === undefined || !headerFormat.test(commentText)) {
+        const trailingNewLineViolation = enforceExtraTrailingLine ? this.doesNewLineEndingViolationExist(text, offset) : false;
+
+        if (commentText === undefined || !headerFormat.test(commentText) || trailingNewLineViolation) {
```

Comment:
If there's a `trailingNewLineViolation`, we should return (before this `if` statement) a `RuleFailure` with a single `Replacement`, which just appends a newline to the end of the valid file header. So this check for `trailingNewLineViolation` will become unnecessary.